financialaid/views.py

```diff
@@ -1,24 +1,217 @@
+        )
+        context["financial_aid_statuses"] = (
+            (status, "Show: {message}".format(message=messages[status]))
+            for status in message_order
+        )
+
+        # Get sort field information
+        sort_link_order = (
+            "first_name",
+            "last_name",
+            "location",
+            "adjusted_cost",
+            "reported_income",
+        )
+        new_sort_direction = "" if self.sort_direction == "-" else "-"
+        context["sort_field_info"] = (
+            {
+                # For appending the sort_by get param on url
+                "sort_field": "{sort_direction}{sort_field}".format(
+                    # If this field is our current sort field, we want to toggle the sort direction, else default ""
+                    sort_direction=new_sort_direction if field == self.sort_field else "",
+                    sort_field=field
+                ),
+                # If this field is the current sort field, we want to indicate the current sort direction
+                "direction_display": self.sort_direction if field == self.sort_field else None,
+                "display": self.sort_fields[field]["display"]
+            }
+            for field in sort_link_order
+        )
+
+        # Required for styling
+        context["style_src"] = get_bundle_url(self.request, "style.js")
+        context["dashboard_src"] = get_bundle_url(self.request, "dashboard.js")
+        js_settings = {
+            "gaTrackingID": settings.GA_TRACKING_ID,
+            "reactGaDebug": settings.REACT_GA_DEBUG,
+            "authenticated": not self.request.user.is_anonymous(),
+            "edx_base_url": settings.EDXORG_BASE_URL,
+        }
+        context["js_settings_json"] = json.dumps(js_settings)
+        return context
+
+    def get_queryset(self):
+        """
+        Gets queryset for ListView to return to view
+        """
+        # Get course price to calculate adjusted cost - we put this first so that we can return
+        # an empty queryset if no valid CoursePrice is found.
+        # Note: This implementation of retrieving a course price is a naive lookup that assumes
+        # all course runs and courses will be the same price for the foreseeable future,
+        # irrespective of the program. Therefore we can just take the price from any currently
+        # enroll-able course run.
+        course_price_object = CoursePrice.objects.filter(
+            is_valid=True,
+            course_run__course__program=self.program
+        ).first()
+        if course_price_object is None:
+            # If course price is not set, we can't meaningfully display any financial aid requests
+            return []
+        else:
+            self.course_price = course_price_object.price
+
+        # Filter by availability and program (self.program set in test_func())
+        financial_aids = FinancialAid.objects.filter(
+            tier_program__program__live=True,
```

Comment:
We should be able to remove these two filters since we already know `self.program` was filtered this way in `test_func`